homeassistant/components/group/sensor.py

```diff
@@ -0,0 +1,405 @@
+        states = []
+        test_states = []
+        last_updated = now(UTC) - timedelta(days=1)
+        sensor_values: list[tuple[str, Any]] = []
+        for entity_id in self._entity_ids:
+            if (state := self.hass.states.get(entity_id)) is not None:
+                states.append(state.state)
+                try:
+                    sensor_values.append((entity_id, float(state.state)))
+                except ValueError:
+                    test_states.append(False)
+                    _LOGGER.warning(
+                        "Unable to use state. Only numerical states are supported, entity %s with value %s excluded from calculation",
+                        entity_id,
+                        state.state,
+                    )
+                    continue
+                test_states.append(True)
+                if state.last_updated > last_updated:
+                    self.last = float(state.state)
+                    self.last_entity_id = entity_id
+
+        # Set group as unavailable if all members are unavailable or missing
+        self._attr_available = any(state != STATE_UNAVAILABLE for state in states)
+
+        valid_state = self.mode(
+            state not in (STATE_UNKNOWN, STATE_UNAVAILABLE) for state in states
+        )
+        valid_state_numeric = self.mode(numeric_state for numeric_state in test_states)
+
+        if not valid_state or not valid_state_numeric:
+            # Set as unknown if any / all member is unknown or unavailable
+            self.min_entity_id = None
+            self.max_entity_id = None
+            self.last_entity_id = None
+            self._attr_native_value = None
+            return
+
+        # Calculate values
+        self._calculate_entity_properties()
+        self._calc_values(sensor_values)
+        self._attr_native_value = getattr(self, self._sensor_attr)
+
+    @property
+    def extra_state_attributes(self) -> dict[str, Any]:
+        """Return the state attributes of the sensor."""
+        attributes: dict[str, Any] = {ATTR_ENTITY_ID: self._entity_ids}
+        if self._sensor_type == "min":
+            attributes[ATTR_MIN_ENTITY_ID] = self.min_entity_id
+        if self._sensor_type == "max":
+            attributes[ATTR_MAX_ENTITY_ID] = self.max_entity_id
+        if self._sensor_type == "last":
+            attributes[ATTR_LAST_ENTITY_ID] = self.last_entity_id
+        return attributes
+
+    @callback
+    def _calc_values(self, sensor_values: list[tuple[str, Any]]) -> None:
+        """Calculate the values."""
+
+        self.min_entity_id, self.min_value = calc_min(sensor_values)
+        self.max_entity_id, self.max_value = calc_max(sensor_values)
+        self.mean = calc_mean(sensor_values, self._round_digits)
+        self.median = calc_median(sensor_values, self._round_digits)
+        self.range = calc_range(sensor_values, self._round_digits)
+        self.sum = calc_sum(sensor_values, self._round_digits)
```

Comment:
Instead of this, maybe:
- Modify all the state calculating functions to return a dict of extra state attributes + native value, including adding a new calculator for `last`
- Add a new member `self._state_calc: Callable[[list[tuple[str, float]], int], tuple[dict[str, str], float | None` which is set to the correct function in `__init__`
- Call it like so: self._extra_state_attribute, self._native_value = self._state_calc(sensor_values, self._round_digits)